src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -2637,6 +2728,21 @@ int RGWMetadataHandlerPut_BucketInstance::put_post(const DoutPrefixProvider *dpp
     }
   } /* update lc */
 
+  /* update bucket topic mapping */
+  {
+    auto* orig_obj = static_cast<RGWBucketInstanceMetadataObject*>(old_obj);
+    auto* orig_bci = (orig_obj ? &orig_obj->get_bci() : nullptr);
+    ret = update_bucket_topic_mappings(dpp, orig_bci, &bci, bihandler->driver);
```

Comment:
this handles operations like PutBucketNotification that overwrite the bucket instance with changes to RGW_ATTR_BUCKET_NOTIFICATION

we need something similar on bucket instance deletion to remove the bucket from any associated topics

Reply:
i was thinking is there a way where ` bucket instance deletion` would be triggered but the `bucket info deletion` will not be triggered, if not then we do not need to handle this in ` bucket instance deletion`

Reply:
"bucket instance" and "bucket info" (aka RGWBucketInfo) are the same thing; is there already some code to remove the mapping on bucket deletion? i didn't see anything. we'll need to handle that both from the DeleteBucket op, and when metadata sync replicates a deletion

Reply:
my bad, i was talking about the bucket_instance and bucket_entry
but now i see what you talking, when the bucket is deleted this [function](https://github.com/ceph/ceph/blob/main/src/rgw/driver/rados/rgw_bucket.cc#L2007) is invoked and we need to call the update_mapping here as well

Reply:
Done adding logic to remove the omap once the bucket is delete. However with the current [issue](https://tracker.ceph.com/issues/20802) where the bucket_instance is not getting deleted on deletion of bucket the mapping is also not getting deleted on secondary site as the deletion is not replicated, but code is added to delete the mapping . Once the tracker is fixed code will also delete the mapping.